lib/diag.c

```diff
@@ -907,14 +907,151 @@ int switchtec_diag_refclk_ctl(struct switchtec_dev *dev, int stack_id, bool en)
 }
 
 /**
- * @brief Get the LTSSM log of a port on a switchtec device
+ * @brief Get the LTSSM log of a port on a gen5 switchtec device
  * @param[in]	dev    Switchtec device handle
  * @param[in]	port   Switchtec Port
  * @param[inout] log_count number of log entries
  * @param[out] log    A pointer to an array containing the log
  *
  */
-int switchtec_diag_ltssm_log(struct switchtec_dev *dev,
+int switchtec_diag_ltssm_log_gen5(struct switchtec_dev *dev,
+	int port, int *log_count,
+	struct switchtec_diag_ltssm_log *log_data)
```

Comment:
We probably can keep switchtec_diag_ltssm_log() and differentiate gen4 and gen5 within it and call xx_log_gen4 and xx_log_gen5 accordingly so that we won't break the API for current gen4 users.